app/services/announcements.ts

```diff
@@ -0,0 +1,85 @@
+import { StatefulService, mutation } from './stateful-service';
+import { UserService } from './user';
+import { HostsService } from './hosts';
+import { Inject } from '../util/injector';
+import { authorizedHeaders } from '../util/requests';
+
+interface IAnnouncementsInfo {
+  id: number;
+  header: string;
+  subHeader: string;
+  link: string;
+  linkTitle: string;
+  thumbnail: string;
+}
+
+export class AnnouncementsService extends StatefulService<IAnnouncementsInfo> {
+  @Inject() private hostsService: HostsService;
+  @Inject() private userService: UserService;
+
+  static initialState: IAnnouncementsInfo = {
+    id: null,
+    header: '',
+    subHeader: null,
+    link: null,
+    linkTitle: null,
+    thumbnail: null
+  };
+
+  async updateBanner() {
+    const newBanner = await this.fetchBanner();
+    this.SET_BANNER(newBanner);
+  }
+
+  bannerExists() {
+    return this.state.id !== null;
+  }
+
+  closeBanner() {
+    this.postBannerClose();
+  }
+
+  private async fetchBanner() {
+    const endpoint = `api/v5/slobs/announcement/get?clientId=${this.userService.getLocalUserId()}`
+    const req = this.formRequest(endpoint);
+    try {
+      const newState = await fetch(req).then((rawResp) => rawResp.json());
```

Comment:
I would recommend using the `handleErrors` helper in `util/requests` to make sure that the request actually succeeded.  The `fetch` API only rejects the promise when there is some sort of network error that prevents the request from going out.  As long as you get a response, the promise is resolved, even if the server returned a 502 for example.  The `handleErrors` helper just makes sure it's a 2xx reponse and rejects the promise otherwise.  More details here: https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch

All that said, I think your code here does do the correct thing in all situations, since in an error scenario, it will either fail to parse the JSON (promise will be rejected), or it will be a JSON object without an `id` in which case the line below will properly handle the error.  So I guess there's no action needed, this is more of an FYI in case you didn't know about that helper.